jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java

```diff
@@ -82,7 +82,8 @@ public void logout(Request request, Response response)
     @Override
     public void setConfiguration(Configuration configuration)
     {
-        _loginService = configuration.getLoginService();
+        if (_loginService == null)
+            _loginService = configuration.getLoginService();
```

Comment:
Should you warn if there are two different non-null login services?

Reply:
I don't think so, this is just so if the `LoginAuthenticator` has been specifically configured with a `LoginService` then we don't want to use the default one from the security handler.

For example you might want to set `HashLoginService` containing different sets of users for different authenticators.